### ISSUE TITLE
Refactor model copy / bbop-manager-minerva 

### ISSUE THREAD

**Issue by kltm** [geneontology/noctua#620]
While we want to get something in so we can iterate on model copy (https://github.com/geneontology/noctua/issues/606 with https://github.com/berkeleybop/bbop-manager-minerva/pull/2) the current implementation has two parts that should be revisited.

The new code pulls in a new promise system and exposes sync/async, something the manager should be abstracting away (for example https://github.com/berkeleybop/bbop-manager-minerva/blob/master/tests/core.test.js#L63 and https://github.com/berkeleybop/bbop-manager-minerva/blob/master/tests/core.test.js#L42). As well, the code seems to introduce a new promise system in addition to the one already in use (https://github.com/berkeleybop/bbop-rest-manager). 

* There should only be one promise system in use (this can be aligned either way)
* The functions should not be sync or async specific.

As we have done other places, a single request can be constructed by iterating over the graph, generating minerva-requests as it goes.

**Comment by kltm** [comment]
It should be noted that if the coy occurs as a Minerva operation, these changes can simply be rolled back, with the new command added to the minerva-requests.

**Comment by kltm** [comment]
@pgaudet I do not believe that this is in scope for the current model copy project.

**Comment by vanaukenk** [comment]
@kltm - can we close this one?

**Comment by kltm** [comment]
There is some old code that I'd like to remove once our minerva-based copying is in place. This is a marker for that, so I'd prefer to keep it open for the moment (or until the roadmap confirms closing our the old manager system).

**Comment by kltm** [comment]
This can be folded into larger refactoring; no need to keep this open for the moment.
